core/controllers/admin_test.py

```diff
@@ -128,12 +128,12 @@ def test_change_rights(self):
         self.signup(
             BOTH_MODERATOR_AND_ADMIN_EMAIL, BOTH_MODERATOR_AND_ADMIN_USERNAME)
 
-        # Navigate to any page. The role is not set.
-        self.testapp.get('/').mustcontain(no=['/moderator', '/admin'])
+        # There should be a redirect on navigating to /
+        self.assertEqual(self.testapp.get('/').status_int, 302)
```

Comment:
Replaced the original line with the one you just posted above.
Also, wrote a new test in base_test.py that checks for redirection in both logged in and logged out states.